overlord/devicestate/devicestate.go

```diff
@@ -526,5 +526,17 @@ func Remodel(st *state.State, new *asserts.Model) ([]*state.TaskSet, error) {
 	}
 	tss = append(tss, state.NewTaskSet(setModel))
 
-	return tss, nil
```

Comment:
after further thinking, because of what we need later, is probably better to achieve this by  turning the inner bits (after the checks basically) of Remodel as it is into remodelTasks, and maybe unit testing that too via a  export_test.go

Reply:
I did that in http://paste.ubuntu.com/p/FKkZMbMTJD/ - is this what you had in mind? You write "after the checks" but at this point we did not do the task wait adjustments yet, this is why I'm wondering.

Reply:
that is a step in the right direction,  my point was to extract remodelTasks from around 435 current state of the file in this branch  (userID := 0), and leave the checks before in Remodel itself. Does it make sense?

Reply:
Aha, sorry for being a bit slow. Now I get it and it makes a lot of sense. Thank you!